shared/src/commonMain/kotlin/org/jetbrains/kotlinconf/TimeProvider.kt

```diff
@@ -38,15 +42,27 @@ class ServerBasedTimeProvider(private val client: APIClient) : TimeProvider {
 }
 
 class FakeTimeProvider(
-    private val fixedTime: GMTDate = GMTDate(
-        year = 2024, month = Month.MAY, dayOfMonth = 23, hours = 13, minutes = 40, seconds = 0
-    )
+    private val baseTime: GMTDate = GMTDate(
+        year = 2024, month = Month.MAY, dayOfMonth = 23, hours = 12, minutes = 40, seconds = 0
+    ),
+    private val freezeTime: Boolean = false,
 ) : TimeProvider {
-    override val time: StateFlow<GMTDate> = MutableStateFlow(fixedTime)
-    override fun now(): GMTDate = fixedTime
+    private val _time = MutableStateFlow(baseTime)
+    override val time: StateFlow<GMTDate> = _time
+    override fun now(): GMTDate = baseTime
     override suspend fun run(): Nothing {
-        while (true) {
-            delay(Long.MAX_VALUE) // This waits forever on its own
+        if (freezeTime) {
+            awaitCancellation()
+        } else {
+            while (true) {
+                // Progress time at 4x speed for testing
+                delay(15.seconds)
+                _time.update { t ->
+                    t.plus(1.minutes).also {
+                        println("Fake time is now $it")
```

Comment:
should we use logger?

Reply:
Since this is only for development purposes, I don't think it's worth it